pkg/processor/runtime/dotnetcore/Wrapper.cs

```diff
@@ -12,31 +12,90 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
-using System;
-using System.Collections.Generic;
-using System.Text;
 using Nuclio.Sdk;
+using System;
 
 namespace processor
 {
 
     public class Wrapper
     {
         private delegate object MethodDelegate(Context context, Event eve);
+        private delegate void InitContextDelegate(Context context);
         private static MethodDelegate methodDelegate;
+        private static string initContextFunctionName = "InitContext";
+        private Type methodType;
 
         private ISocketHandler socketHandler;
+        private Context context;
 
         public Wrapper(string dllPath, string typeName, string methodName, string socketPath)
         {
+            
             CreateTypeAndFunction(dllPath, typeName, methodName);
-            StartUnixSocketHandler(socketPath);
+
+            InitUnixSocketHandler(socketPath);
+
+            //Run the function's method that initializes the context
+            try
+            {
+                ExecuteInitContext();
+                socketHandler.SendMessage("{ 'kind': 'wrapperInitialized', 'attributes': {'ready': 'true'}");
+                context.Logger.Info("Wrapper ready.");
+
+            }
+            catch (Exception e)
+            {
+                context.Logger.Error(e.Message);
+                context.Logger.Error(e.StackTrace);
+                throw new Exception(e.Message);
+            }
+
+            StartUnixSocketHandler();
+
+        }
+
+        private void ExecuteInitContext()
+        {
+            if (methodType == null)
+            {
+                context.Logger.Error("Failed to execute InitContext(): type not found.");
+                throw new Exception("Failed to execute " + initContextFunctionName + ": type not found.");
+            }
```

Comment:
I dont see the relation accurate. may you explain the relation between methodType here to initcontext specifically?

Reply:
My thinking here was that if we cannot locate the class that contains the function, then we'd be in an invalid state because we would not be able to invoke the function or `InitContext()`. So we'd just throw an exception because the function specification is incorrect. 

I think we need a null reference check somewhere so that we don't call reflected methods on a null object, which could result from the function specification being incorrect (e.g., Reflection can't find the class or methods).